src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+        private static class SslProvider
+        {
+            private delegate int SslCtxCallback(IntPtr curl, IntPtr sslCtx, IntPtr userPtr);
+
+            private static readonly SslCtxCallback s_sslCtxCallback = SetSslCtxVerifyCallback;
+            private static readonly Interop.Ssl.AppVerifyCallback s_sslVerifyCallback = VerifyCertChain;
+
+            internal static void SetSslOptions(EasyRequest easy)
+            {
+                int answer = Interop.libcurl.curl_easy_setopt(
+                    easy._easyHandle,
+                    Interop.libcurl.CURLoption.CURLOPT_SSL_CTX_FUNCTION,
+                    s_sslCtxCallback);
+
+                switch (answer)
+                {
+                    case Interop.libcurl.CURLcode.CURLE_OK:
+                        break;
+                    case Interop.libcurl.CURLcode.CURLE_NOT_BUILT_IN:
+                        VerboseTrace("CURLOPT_SSL_CTX_FUNCTION is not supported, platform default https chain building in use");
+                        break;
+                    default:
+                        ThrowIfCURLEError(answer);
+                        break;
+                }
+            }
+
+            private static int SetSslCtxVerifyCallback(
+                IntPtr curl,
+                IntPtr sslCtx,
+                IntPtr userPtr)
+            {
+                using (SafeSslContextHandle ctx = new SafeSslContextHandle(sslCtx, ownsHandle: false))
+                {
+                    Interop.Ssl.SslCtxSetCertVerifyCallback(ctx, s_sslVerifyCallback, userPtr);
+                }
+
+                return Interop.libcurl.CURLcode.CURLE_OK;
+            }
+
+            private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
```

Comment:
One doubt about the verification - does the storeCtx here corr. to /etc/ssl/certs? What I am trying to confirm is if the verification fails because CA cert is missing on the machine and then user does a "sudo update-ca-certiicates", things will work fine, correct?